stdlib/3/tempfile.pyi

```diff
@@ -17,6 +18,11 @@ TMP_MAX: int
 tempdir: Optional[str]
 template: str
 
+if sys.version_info >= (3, 6):
+    _DirT = Union[AnyStr, os.PathLike[AnyStr]]
+else:
+    _DirT = Union[AnyStr]
```

Comment:
If you want to use this trick then you must to write `_DirT[AnyStr]` everywhere, otherwise it will expand to `Any` (like any other missing type argument to generic).